File: killrvideo-service-users/src/main/java/com/killrvideo/service/user/dao/UserDaoConfig.java
package com.killrvideo.service.user.dao;

import com.datastax.oss.driver.api.core.CqlSession;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class UserDaoConfig {
    @Bean
    public UserMapper userMapper(CqlSession session) {
        return UserMapper.builder(session).build();
    }
}

File: killrvideo-service-users/src/main/kotlin/com/killrvideo/service/user/dao/UserDaoConfig.kt
package com.killrvideo.service.user.dao

import com.datastax.oss.driver.api.core.CqlSession
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class UserDaoConfig {

    @Bean
    open fun userMapper(session: CqlSession): UserMapper {
        return UserMapperBuilder(session).build()
    }
}
